serverLib.sml

```diff
@@ -666,21 +675,39 @@ in
       val typ = read_job_type inp
                 handle IO.Io _ => cgi_die 500 ["cannot open ",f]
                      | Option => cgi_die 500 [f," has invalid file format"]
+      val worker = read_job_worker inp
       val format_type = if q = "finished" then span (status_attrs (read_status inp)) else String.concat
-      val last_date = if q = "running" then read_last_date inp else NONE
       val () = TextIO.closeIn inp
+      val inp2 = TextIO.openIn f
```

Comment:
I think I'd rather this be called `inp` again. I guess you've thought through whether this could be done with a single pass through the file (and it can't)?

Reply:
Yes, Ideally a single function should read all the information that can possible be in a job file and store that in a record, however the functions currently in place to do this are scattered between `serverLib.sml` and `apiLib.sml`  and in a bit tricky to put everything in one place, so I guess that for now this is the best option